cjworkbench/i18n/trans.py

```diff
@@ -46,37 +32,20 @@ def trans(message_id, *, default, context=None, parameters={}):
     For code parsing reasons, respect the following order when passing keyword arguments:
         `message_id` and then `default` and then `context` and then everything else
     """
-    return do_trans(
-        get_language(),
-        message_id,
-        default=default,
-        context=context,
-        parameters=parameters,
-    )
-
-
-def do_trans(locale_id, message_id, *, default, context=None, parameters={}):
-    """Translate the given message ID to the given locale
-    
-    HTML is not escaped.
-    
-    For code parsing reasons, respect the following order when passing keyword arguments:
-        `message_id` and then `default` and then `context` and then everything else
-    """
-    return _get_translations(locale_id).trans(
+    return _get_translations(get_language()).trans(
         message_id, default=default, context=context, parameters=parameters
     )
 
 
-def trans_html(locale, message_id, *, default, context=None, parameters={}, tags={}):
-    """Translate the given message ID to the current locale
+def trans_html(locale_id, message_id, *, default, context=None, parameters={}, tags={}):
```

Comment:
Thanks for simplifying! Now that the functions are right next to each other, we can see a consistency problem: `trans_html()` takes a `locale_id` parameter, but `trans()` doesn't. These two functions are beside each other, so they ought to have similar signatures. How can we make that happen?

Reply:
I don't think we can make that happen. `trans_html` can be used in cases where the django locale is unset or wrong, for example when we send an email that data has been updated, hence it needs a locale parameter. On the other hand, `trans` is for cases where the locale of django can (or must) be trusted; currently, it is only used via `trans_lazy` to define placeholders in the signup form, which have to be locale-agnostic.

Reply:
We could remove `trans` altogether and only keep `trans_lazy`...

Reply:
Thanks for explaining. I think long-term we'll want to make these signatures consistent; but seeing as it's difficult, I'm happy to merge as-is.